subsys/bluetooth/host/l2cap.c

```diff
@@ -1003,14 +1004,26 @@ static uint16_t l2cap_chan_accept(struct bt_conn *conn,
 	return BT_L2CAP_LE_SUCCESS;
 }
 
-static bool l2cap_check_security(struct bt_conn *conn,
+static int l2cap_check_security(struct bt_conn *conn,
 				 struct bt_l2cap_server *server)
 {
 	if (IS_ENABLED(CONFIG_BT_CONN_DISABLE_SECURITY)) {
-		return true;
+		return 0;
 	}
 
-	return conn->sec_level >= server->sec_level;
+	if (conn->sec_level < server->sec_level) {
+		if (server->sec_level == BT_SECURITY_L2) {
+			return BT_L2CAP_LE_ERR_ENCRYPTION;
```

Comment:
From what I could tell from the spec then returning BT_L2CAP_LE_ERR_AUTHENTICATION was alright for this case as well.
```
Note: When encryption is not enabled, the result value “Connection refused –
insufficient authentication” does not indicate that MITM protection is required.
```

Reply:
Good point. I browsed through L2CAP API and this should be Authentication errors, as it was before commit. Authorization error should be returned if server accept callback returns -EACCES. I updated the PR